packages/react-testing/src/matchers/utilities.ts

```diff
@@ -57,6 +57,49 @@ export function assertIsNode(
   }
 }
 
+export function diffs(
+  element: Element<any>[],
+  props: object,
+  expand?: boolean,
+) {
+  return element.reduce<string>(
+    (diffs, element, index) =>
+      `${diffs}${index === 0 ? '' : '\n\n'}${normalizedDiff(element, props, {
+        expand,
+        showLegend: index === 0,
+      })}`,
+    '',
+  );
+}
+
+function normalizedDiff(
+  element: Element<any>,
+  props: object,
+  {expand = false, showLegend = false},
+) {
+  const result =
+    diffPropsForNode(element, props, {
+      expand,
+    }) || '';
+
+  return showLegend ? result : result.split('\n\n')[1];
+}
+
+export function printType(type: string | React.ComponentType<any>) {
+  if (typeof type === 'object' && '_context' in type) {
+    const context = (type as any)._context;
+    const componentName = type === context.Provider ? 'Provider' : 'Consumer';
+    const displayName = context.displayName || 'Context';
+    return `<${displayName}.${componentName} />`;
```

Comment:
Neat, this is the part that fixes `<undefined />` I guess? I find these intermediary variables also much nicer than the ternary inline below, could we make it consistent?